src/app_modes/send_pose.rs

```diff
@@ -0,0 +1,217 @@
+    }
+    fn handle_input(&mut self, input: &String) {
+        self.viewport.borrow_mut().handle_input(input);
+        match input.as_str() {
+            input::UP => self.move_new_pose(self.increment, 0.0, 0.0),
+            input::DOWN => self.move_new_pose(-self.increment, 0.0, 0.0),
+            input::LEFT => self.move_new_pose(0.0, self.increment, 0.0),
+            input::RIGHT => self.move_new_pose(0.0, -self.increment, 0.0),
+            input::ROT_LEFT => self.move_new_pose(0.0, 0.0, self.increment),
+            input::ROT_RIGHT => self.move_new_pose(0.0, 0.0, -self.increment),
+            input::INC_STEP => self.increment += 0.1,
+            input::DEC_STEP => self.increment -= 0.1,
+            input::CANCEL => self.reset(),
+            input::CONFIRM => self.send_new_pose(),
+            _ => (),
+        }
+    }
+
+    fn get_name(&self) -> String {
+        "Send Pose".to_string()
+    }
+
+    fn get_description(&self) -> Vec<String> {
+        vec![
+            "This mode allows to publish a PoseStamped message on a topic.".to_string(),
+            "The viewport is centered on the preview outline of where the pose is on the map."
+                .to_string(),
+        ]
+    }
+
+    fn get_keymap(&self) -> Vec<[String; 2]> {
+        let mut keymap = vec![
+            [
+                input::UP.to_string(),
+                "Shifts the pose estimate positively along the x axis.".to_string(),
+            ],
+            [
+                input::DOWN.to_string(),
+                "Shifts the pose estimate negatively along the x axis.".to_string(),
+            ],
+            [
+                input::RIGHT.to_string(),
+                "Shifts the pose estimate positively along the y axis.".to_string(),
+            ],
+            [
+                input::LEFT.to_string(),
+                "Shifts the pose estimate negatively along the y axis.".to_string(),
+            ],
+            [
+                input::ROT_LEFT.to_string(),
+                "Rotates the pose estimate counter-clockwise.".to_string(),
+            ],
+            [
+                input::ROT_RIGHT.to_string(),
+                "Rotates the pose estimate clockwise.".to_string(),
+            ],
+            [
+                input::CANCEL.to_string(),
+                "Resets the pose estimate.".to_string(),
+            ],
+            [
+                input::CONFIRM.to_string(),
+                "Sends the pose estimate.".to_string(),
+            ],
+            [
+                input::INC_STEP.to_string(),
+                "Increases the step size for manipulating the pose estimate.".to_string(),
+            ],
+            [
+                input::DEC_STEP.to_string(),
+                "Decreases the step size for manipulating the pose estimate.".to_string(),
+            ],
+        ];
+        keymap.extend(self.viewport.borrow().get_keymap());
+        keymap
+    }
+}
+
+impl UseViewport for SendPose {
```

Comment:
something changed here relative to main: the ghost robot is drawn by default when entering the mode.this is problematic when the robot moves a little

Reply:
I tried to catch every bug here but apparently I missed some. Will fix